### DISCUSSION TITLE
How to create installable Eclipse-CS snapshot plugin artifacts?

### DISCUSSION THREAD

**Discussion by rnveach** [checkstyle/eclipse-cs#473] [Q&A]
I run `mvn clean package` on master and get a successful build. I try to go to Eclipse and install the plugin from `eclipse-cs\net.sf.eclipsecs-updatesite\target\repository` but it gets the following error:

````
Cannot complete the install because one or more required items could not be found.
  Software being installed: Eclipse Checkstyle Plug-in 10.4.0.202302051801 (net.sf.eclipsecs.feature.group 10.4.0.202302051801)
  Missing requirement: Eclipse Checkstyle 10.4.0.202302051801 (net.sf.eclipsecs.core 10.4.0.202302051801) requires 'java.package; org.dom4j 2.1.3' but it could not be found
````

When comparing my packaged version versus the release, I notice that the core JAR is missing the 3rd party JARs (dom4j, snakeyml, javaassist). I assume this is why Eclipse fails to install, but I am not seeing what is missing to include them.

I also noticed my doc JAR is 1000x larger than the release version, but I assume this is unrelated.

This is the same with `net.sf.eclipsecs-updatesite/target/net.sf.eclipsecs-updatesite_10.4.0.202302051801.zip`.

I have no issues compiling the project in Eclipse (no errors).

**Comment by rnveach** [comment]
Moved to https://github.com/checkstyle/eclipse-cs/issues/475#issue-1571619891 when I found the same instructions to create the plugin and could isolate when the issue seems to have started.

**Reply by Bananeweizen** [reply]
The update site packaging didn't use a (recommended) setting for including transitive dependencies. You would have seen the same error with older installations when unchecking "Contact all update sites..." during installation, since even before my refactoring the update site was not fully self-contained.
